lib/core/utils/matches.js

```diff
@@ -240,12 +243,36 @@ export function matchesExpression(vNode, expressions, matchAnyParent) {
 
     matches =
       matches &&
-      matchesExpression(vNode.parent, exps, expression.combinator === ' ');
+      optimizedMatchesExpression(
+        vNode.parent,
+        expressions,
+        index - 1,
+        expression.combinator === ' '
+      );
   }
 
   return matches;
 }
 
+/**
+ * Determine if a virtual node matches a Slick format CSS expression
+ *
+ * @private
+ * @method matchesExpression
+ * @memberof axe.utils
+ * @param {VirtualNode} vNode VirtualNode to match
+ * @param {Object|Object[]} expressions CSS selector expression or array of expressions
+ * @returns {Boolean}
+ */
+export function matchesExpression(vNode, expressions, matchAnyParent) {
+  return optimizedMatchesExpression(
+    vNode,
+    expressions,
+    expressions.length - 1,
```

Comment:
This will return `NaN` if `expressions` is an object instead of an array. Since this didn't error then we may be missing a test case somewhere.

```suggestion
    typeof expressions.length === 'number' ? expressions.length - 1 : 0,
```

Reply:
This is intended, as the argument is ignored if `expressions` is not an array (JSDoc states that the type of `index` is `Number|NaN`). Can do a check and pass `null` if you think that's preferable.